server.js

```diff
@@ -2,37 +2,162 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import showData from './data/netflix-titles.json'
+import { Shows } from './models'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+const dotenv = require('dotenv')
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const path = require('path')
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.sendFile(path.join(__dirname+'/public/index.html'))
 })
 
-// Start the server
+app.get('/shows', async (req, res) => {   
+  const { title, year, country, actor } = req.query
+  let myFilter = {}
+
+  if (title) {
+    myFilter['title'] = new RegExp(`${title}`, 'i')
+  }
+  
+  if (year) {
+    myFilter['release_year'] = year;
+  }
+
+  if (country) {
+    myFilter['country'] = new RegExp(`${country}`, 'i')
+  }
+
+  if (actor) {
+    myFilter['cast'] = new RegExp(`${actor}`, 'i')
+  }
+  
+  const show = await Shows.find(myFilter).collation({ locale: 'en_US', strength: 1 });
+  
+  const page = +(req.query.page ?? 1)
+  const pageSize = +(req.query.pageSize ?? 20)
+  const startIndex = (page -1) * pageSize
+  const endIndex = startIndex + pageSize 
+  const filteredShows = show.slice(startIndex, endIndex)
+  const returnObject = { 
+    pageSize: pageSize,
+    page: page,
+    maxPages: Math.ceil(show.length/pageSize),
+    totalShows: show.length,
+    results: filteredShows
```

Comment:
You created pagination using the slice method on a very big set of data already fetched from the database. Maybe it would be better to Fetch only part of date we need now from the database. Use mongoose methods like `skip()`, `limit()` and `sort()`.